packages/react-components/src/components/ChatMessageComponent.tsx

```diff
@@ -34,217 +19,42 @@ type ChatMessageProps = {
+            props.message.messageId &&
+            (await props.onUpdateMessage(props.message.messageId, text));
           setIsEditing(false);
         }}
         onCancel={() => {
           setIsEditing(false);
         }}
       />
     );
-  }
-  const messageContentItem = GenerateMessageContent(message, strings.liveAuthorIntro);
-
-  const chatMessage = (
-    <div {...longPressProps}>
-      <Chat.Message
-        className={mergeStyles(messageContainerStyle as IStyle)}
-        styles={messageContainerStyle}
-        content={messageContentItem}
-        author={<Text className={chatMessageDateStyle}>{message.senderDisplayName}</Text>}
-        mine={message.mine}
-        timestamp={
-          <Text data-ui-id={ids.messageTimestamp}>
-            {message.createdOn
-              ? showDate
-                ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
-                : formatTimeForChatMessage(message.createdOn)
-              : undefined}
-          </Text>
-        }
-        details={
-          message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
-        }
-        positionActionMenu={false}
-        actionMenu={!disableEditing && message.status !== 'sending' && message.mine ? actionMenu : undefined}
-      />
-    </div>
-  );
-
-  return chatMessage;
-};
-
-const MoreMenu = ({
-  hideContextMenu,
-  onEditClick,
-  onRemoveClick,
-  onDismiss,
-  strings
-}: {
-  hideContextMenu: boolean;
-  onEditClick: () => void;
-  onRemoveClick: () => void;
-  onDismiss: () => void;
-  strings: MessageThreadStrings;
-}): JSX.Element => {
-  const menuRef = useRef<HTMLDivElement | null>(null);
-
-  const menuItems = useMemo(
-    (): IContextualMenuItem[] => [
-      {
-        key: 'Edit',
-        text: strings.editMessage,
-        iconProps: { iconName: 'MessageEdit', styles: menuIconStyleSet },
-        onClick: onEditClick
-      },
-      {
-        key: 'Remove',
-        text: strings.removeMessage,
-        iconProps: {
-          iconName: 'MessageRemove',
-          styles: menuIconStyleSet
-        },
-        onClick: onRemoveClick
-      }
-    ],
-    [onEditClick, onRemoveClick, strings]
-  );
-
-  return (
-    <div ref={menuRef}>
-      <MoreIcon
-        className={iconWrapperStyle}
-        {...{
-          outline: true
-        }}
-      />
-      <ContextualMenu
-        items={menuItems}
-        hidden={hideContextMenu}
-        target={menuRef}
-        onDismiss={() => onDismiss()}
-        directionalHint={DirectionalHint.bottomLeftEdge}
-        className={chatMessageMenuStyle}
+  } else {
+    const onRemoveClick = async (): Promise<void | undefined> =>
+      props.onDeleteMessage && props.message.messageId
+        ? await props.onDeleteMessage(props.message.messageId)
+        : undefined;
+    return (
+      <ChatMessageComponentAsMessageBubble
+        {...props}
+        onRemoveClick={onRemoveClick}
```

Comment:
Please `useCallback` for both `onRemoveClick` and `onEditClick` to avoid spurious renders.

There are many messages in the `MessageThread`. This is the sort of component where spurious renders are really bad.

Reply:
Updated